xds/internal/xdsclient/xds.go

```diff
@@ -360,6 +366,58 @@ func generateRDSUpdateFromRouteConfiguration(rc *v3routepb.RouteConfiguration, l
 	return RouteConfigUpdate{VirtualHosts: vhs}, nil
 }
 
+func generateRetryConfig(rp *v3routepb.RetryPolicy) (*RetryConfig, error) {
+	if !env.RetrySupport || rp == nil {
+		return nil, nil
+	}
+
+	cfg := &RetryConfig{RetryOn: make(map[codes.Code]bool)}
+	for _, s := range strings.Split(rp.GetRetryOn(), ",") {
+		switch strings.TrimSpace(strings.ToLower(s)) {
+		case "cancelled":
+			cfg.RetryOn[codes.Canceled] = true
+		case "deadline-exceeded":
+			cfg.RetryOn[codes.DeadlineExceeded] = true
+		case "internal":
+			cfg.RetryOn[codes.Internal] = true
+		case "resource-exhausted":
+			cfg.RetryOn[codes.ResourceExhausted] = true
+		case "unavailable":
+			cfg.RetryOn[codes.Unavailable] = true
+		}
+	}
+	if len(cfg.RetryOn) == 0 {
+		return &RetryConfig{}, nil
```

Comment:
Any reason you return a RetryConfig with a nil map vs. return cfg with the initialized map of length 0?

Reply:
Also, Mark has a question in document about validating the other fields even if no policies that are supported in GetRetryOn. Make sure we don't want to validate the other fields.

Reply:
"What should we do if retry_on contains no policies that we support?  I assume that we should just not return any retry policy in that case.  But should we still validate the other fields in that case?" "Actually, if retry_on does not contain any policies that we support but per_try_timeout is set, then we still need to return a policy." - Mark Roth in Google doc. From the second sentence it seems your commit adding validations even if no policies supported is definitely the correct change.

Reply:
Noted about map, thanks for the explanation.